integration/test_overload_protection.py

```
@@ -0,0 +1,184 @@
+
+        # Disabled when limit=0.
+        client.execute_command("CONFIG", "SET",
+                               "search.max-query-queue-depth", "0")
+        client.execute_command(
+            "FT._DEBUG", "CONTROLLED_VARIABLE", "SET",
+            "ForceQueueDepthExceeded", "yes")
+        try:
+            client.execute_command(
+                "FT.SEARCH", "idx", "@tag1:{common}", "LIMIT", "0", "1")
+        except ResponseError as e:
+            assert False, f"Should not reject when limit=0: {e}"
```

Ruff (0.16.0)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)